File: TownOfSalemCalculator/src/townofsalemcalculator/Conditions/Concrete/KillMessages/Cleaned.java
package townofsalemcalculator.Conditions.Concrete.KillMessages;

import java.util.Arrays;
import java.util.List;
import static townofsalemcalculator.Role.Janitor;

/**
 * The Concrete Condition that someone was cleaned while he was killed
 * @author Multifacio
 * @version 1.1
 * @since 2017-2-10
 */
public class Cleaned extends KillMessage {
    
    public Cleaned() {
        super(Janitor);
    }
    
    @Override
    public String getCondition() {
        return "Someone was cleaned (we could not determine its role)";
    }

    @Override
    public List<String> keyWords() {
        List<String> keyWords = Arrays.asList(new String[]{"We", "Could", "Not", "Determine", "Role", "Cleaned"});
        keyWords.addAll(Janitor.getKeyWords());
        return keyWords;
    }
}

File: src/Conditions/Concrete/KillTypes/Cleaned.py
from src.Concepts.Role import Role
from src.Conditions.Concrete.ExistenceCondition import ExistenceCondition
from typing import FrozenSet

class Cleaned(ExistenceCondition):
    """ The Cleaned condition is used if some player died and was cleaned (by a Janitor). """

    def __init__(self, amnesiacRemembered: FrozenSet[Role] = None):
        super().__init__(frozenset({Role.JANITOR}), amnesiacRemembered)